firefox-ios/firefox-ios-tests/Tests/ClientTests/Mocks/MockWindowManager.swift

```diff
@@ -27,7 +27,7 @@ final class MockWindowManager: WindowManager {
     }
 
     func tabManager(for windowUUID: WindowUUID) -> TabManager {
-        wrappedManager.tabManager(for: windowUUID)
+        return MockTabManager()
```

Comment:
@mattreaganmozilla I had to mock this out since tests were failing because `tabManager` was `nil`, which triggered the `assertionFailure`

![image](https://github.com/user-attachments/assets/4f098fe9-ab6f-41c5-bec1-422734a1dc02)

I was curious on why we are using the `WindowManagerImplementation` for tests, but returning the mock seems to fix the issue. Let me know if you have any concerns!